### ISSUE TITLE
Unable to locate the system's glibc - Raspbian Jessie

### ISSUE THREAD

**Issue by AlexisMeza** [Linuxbrew/linuxbrew#905]
I ran brew update twice and brew doctor. I tried this on both a Raspberry 1 Model B (RP1B) and Raspberry Pi 2 Model B (RP2B). RP2B is a clean 2016-02-09 Raspbian install with all apt-get update/updates/dist-upgrade applied while RP1B is using Raspbian that has been upgraded to Jessie over time. 

I tried:
`brew install nginx --devel`
Brew begins to install pcre dependency and stops with "Unable to locate the system's glibc". I have tried clearing the cache, setting export HOMEBREW_BUILD_FROM_SOURCE=1 on my .zshrc file, sourcing it and/or logging in/out and checking it with the following:
`echo $(brew --env)`

Each time I try to install nginx, I have the following:
> Last 15 lines from /home/xxxxxx/.cache/Homebrew/Logs/pcre/01.configure:
> checking whether build environment is sane... yes
> checking for a thread-safe mkdir -p... /bin/mkdir -p
> checking for gawk... gawk
> checking whether make sets $(MAKE)... yes
> checking whether make supports nested variables... yes
> checking whether make supports nested variables... (cached) yes
> checking for style of include used by make... GNU
> checking for gcc... /usr/bin/gcc-4.9
> checking whether the C compiler works... yes
> checking for C compiler default output file name... a.out
> checking for suffix of executables... 
> checking whether we are cross compiling... configure: error: in `/tmp/pcre20160310-3078-ofqrqi/pcre-8.38':
> configure: error: cannot run C compiled programs.
> If you meant to cross compile, use `--host'.
> See `config.log' for more details

On my RP1B, I have used linuxbrew for months without issues. I tried creating a Debian x64 VM and had no issues installing nginx or any other formula.

Running brew doctor after a failed install returns:
> Warning: Broken symlinks were found. Remove them with `brew prune`:
>     /home/xxxxxx/.linuxbrew/lib/ld.so

ls -Alh that file returns
> lrwxrwxrwx 1 xxx xxx 27 Mar 10 19:01 /home/xxxxxx/.linuxbrew/lib/ld.so -> /lib64/ld-linux-x86-64.so.2

I have tried running brew prune which removes the link but trying to install nginx fails again with the same issue described above.

Thanks all!

**Comment by sjackman** [comment]
You'll have to fix this line for 32-bit Linux. 32-bit support is still experimental.
https://github.com/Linuxbrew/linuxbrew/blob/3b0d037062c47d22c4e1c327f0edf14ad11e0d38/Library/Homebrew/requirements/glibc_requirement.rb#L16
```ruby
libc = ["/lib/x86_64-linux-gnu/libc.so.6", "/lib64/libc.so.6"].find do |s|
```

**Comment by sjackman** [comment]
Adding `/lib/libc.so.6` to this list would probably fix it.

**Comment by sjackman** [comment]
Did that fix it for you, Alexis?

**Comment by sjackman** [comment]
I'm glad to hear that Linuxbrew on Raspberry Pi has been working for you! Although I do own a Raspberry Pi, I don't get to play around with it too often.

**Comment by sjackman** [comment]
There may be other fixes needed. This'll get you closer though, I hope.

**Comment by AlexisMeza** [comment]
Shaun,

Thanks for that push! I'll make a pull request to make the adjustments unless someone else gets to it. Here are my findings:

glibc_requirement.rb - added "/lib/arm-linux-gnueabihf/libc.so.6"
`libc = ["/lib/x86_64-linux-gnu/libc.so.6", "/lib64/libc.so.6", "/lib/arm-linux-gnueabihf/libc.so.6"].find do |s|`

Next, I focused on that comes up on brew doctor so I searched for it:
`grep -r "ld-linux-x86-64.so.2" ~/.linuxbrew/Library`

I made adjustments to Formula/glibc.rb and Homebrew/cmd/install.rb:~197
`interpreter = glibc && glibc.installed? ? glibc.lib/"ld-linux.so.3" : "/lib/ld-linux.so.3"`
Obviously, this line removes support for 64-bit. I'll have to look into Linuxbrew to see if there is a check for 64-bit and apply it here. I'm happy to do the legwork sometime this weekend but again, I thought I'd share this here if someone is looking for a solution or if someone has time to make the changes before I get to them.

Besides this minor issue, I've been using Linuxbrew on the Raspberry Pi without issues since you added support for it last summer.

**Comment by sjackman** [comment]
A pull request would be much appreciated. Thanks, Alexis.

**Comment by sjackman** [comment]
I'll break out my Rasberry Pi and give it a whirl.

**Comment by gpotter2** [comment]
Hum, this is an old thing, right. Small note:

> /lib/libc.so.6

Is on many linux 32bits versions:

`/lib/i386-linux-gnu/libc.so.6`

That should be taken in account

**Comment by sjackman** [comment]
Both of these paths are on the search list:
```ruby
    libc = ["/lib/x86_64-linux-gnu/libc.so.6", "/lib64/libc.so.6", "/lib/libc.so.6", "/lib/i386-linux-gnu/libc.so.6", "/lib/arm-linux-gnueabihf/libc.so.6"].find do |s|
```
https://github.com/Linuxbrew/linuxbrew/blob/master/Library/Homebrew/requirements/glibc_requirement.rb#L16

**Comment by PDDStudio** [comment]
As because this issue helped me to get the solution for my problem, I thought I'd append this here:

I had the same error as mentioned above (problems with finding `libc` and `ld`) when I was about to install linuxbrew. While validating the paths, I saw that the array you ( @sjackman  ) didn't include locations for my 64bit debian jessy build.

You mentioned above that this step is actually required for 32bit devices only, but in my case (I'm running a bootstrapped debian-jessy) it seems like the the system couldn't find the static libraries - even though they're compiled for a 64bit architecture. 

Appending `/lib/aarch64-linux-gnu/libc.so.6` in `glibc_requirement.rb` :

```bash
libc = ["/lib/aarch64-linux-gnu/libc.so.6", "/lib/x86_64-linux-gnu/libc.so.6", "/lib64/libc.so.6", "/lib/libc.so.6", "/lib/i386-linux-gnu/libc.so.6", "/lib/arm-linux-gnueabihf/libc.so.6"].find do |s|
```

and `/lib/aarch64-linux-gnu/ld-2.19.so` (the version I installed - can be different) in `cmd/install.rb` :

```bash
sys_interpreter = ["/lib/aarch64-linux-gnu/ld-2.19.so", "/lib64/ld-linux-x86-64.so.2", "/lib/ld-linux.so.3", "/lib/ld-linux.so.2", "/lib/ld-linux-armhf.so.3"].find do |s|
```

finally solved my errors when running `brew`.

Another *(maybe important)* thing is that linuxbrew itself created an symlink for `lib64 -> lib` inside `.linuxbrew` internally, so for me it seems like the bootstrapped image caused this? (Correct me if I'm wrong)

Maybe this can be respected in future versions of `Linuxbrew` as well (checking for 'non-default'-64bit distributions as well instead of only 32bit in `/lib`).

For now I just wanted to collect my knowledge somewhere in case someone else runs into the same problems as I did.

Finally, I can confirm that linuxbrew is running on an Android-Smartphone (both - armhf/arm64) inside a bootstrapped debian-system deployed with chroot.

Regards
